### PR TITLE
vmpool: improve test time

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
This series of commits gradually remove complexity of pool tests to the point where we reduce the runtime from ~19 seconds down to ~250ms. On top of this we remove some unnecessary code and fix few issues.

### Special notes for your reviewer
These changes follow the logic introduced here https://github.com/kubevirt/kubevirt/pull/12468

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
/hold
Based on https://github.com/kubevirt/kubevirt/pull/12505

**Review by 0xFelix — APPROVED** [review]
Would have been nice if the rename to `fakeVirtClient` could have been avoided but LGTM otherwise!

/lgtm

File: pkg/virt-controller/watch/pool_test.go
Comment (xpivarc) [inline pkg/virt-controller/watch/pool_test.go:658]: say goodbye to "flaky by nature"

File: pkg/virt-controller/watch/pool_test.go
Comment (xpivarc) [inline pkg/virt-controller/watch/pool_test.go:110]: Can we remove the `expectQueue`?

**Review by xpivarc — COMMENTED** [review]
/approve
/hold
Feel free to unhold as my comment is more of nit

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**
